### PR TITLE
More include cleanups

### PR DESCRIPTION
**Pull request by MaxKellermann**

### CODE REVIEW

**Comment by notspiff** [comment]
real build errors.

**Comment by MaxKellermann** [comment]
Oh, I'll fix those later (I didn't build those operating systems). This is a problem which was already there before, just wasn't visible because there were too many includes.

**Comment by notspiff** [comment]
i know. just wanted to point out it's not a false alarm thingie.

**Comment by MaxKellermann** [comment]
Pushed again with a one-line fix (amended the ActorProtocol commit) and tested with the Android build. Let's see what Jenkins about it.

**Review by notspiff — APPROVED** [review]

File: xbmc/utils/ActorProtocol.cpp
Comment (pkerling) [inline xbmc/utils/ActorProtocol.cpp]: Is this a system header? What is it for?

Comment (a1rwulf) [reply xbmc/utils/ActorProtocol.cpp]: Afaik it's the systemheader holding memset, memcpy, etc.
Given the fact that <memory> is already included in ActorProtocol.h now, I think this is not necessary at all.

Comment (pkerling) [reply xbmc/utils/ActorProtocol.cpp]: If so, it should be `<cstring>`. `<memory>` is C++ stuff and unrelated.

Comment (a1rwulf) [reply xbmc/utils/ActorProtocol.cpp]: yeah, my bad <memory> != <memory.h>.
In this case it's not unnecessary ofc.
After some short googling I found memory.h being something rather old and <cstring> seems to be the better choice.
Anyway we've been using "memory.h" in the ActorProtocl header before, so I won't block this PR from being merged because of this minor.

Comment (pkerling) [reply xbmc/utils/ActorProtocol.cpp]: @MaxKellermann would be super nice if you could correct this in the course of include cleanup but it's also not a blocker for me

Comment (MaxKellermann) [reply xbmc/utils/ActorProtocol.cpp]: Will do.

**Comment by MaxKellermann** [comment]
Rebased, fixed `"memory.h"` (replaced with `<cstring>` as suggested), and fixed a new breakage in `ProfilesManager.cpp`.

**Comment by pkerling** [comment]
Thank you!

**Pull request merged**
